### PR TITLE
DCOS-19580: Add DSS volumes to pods.

### PR DESCRIPTION
**Pull request by Poltergeist**
This will add the support for DSS volumes for Pods
 ---
Add to your Config.dev.js

```
dss: { enabled: true }
```

### CODE REVIEW

**Comment by nLight** [comment]
So it's GiB for pods and MiB for singles?
screen shot 2018-01-04 at 15 50 08
screen shot 2018-01-04 at 15 50 00

**Comment by Poltergeist** [comment]
@nLight At least this is what the prototype said.

**Comment by nLight** [comment]
@Poltergeist shall we ask someone from the storage team to be sure? I wouldn't trust design mockups.

**Comment by juliangieseke** [comment]
I remember having some issues with this when writing the system - we should
be able to check them for the correct text.
Dmitriy Rozhkov <notifications@github.com> schrieb am Do. 4. Jan. 2018 um
17:36:

**Comment by juliangieseke** [comment]
Fails are flaky ones
Alfred test this

**Comment by juliangieseke** [comment]
Alfred test this

**Comment by nLight** [comment]
Alfred test this

**Comment by Poltergeist** [comment]
Local volumes are in Mega byte external ones in Gigabyte

**Comment by Poltergeist** [comment]
I addressed the Issue that @nLight was experiencing.

**Comment by nLight** [comment]
Alfred test this

**Comment by bstavroulakis** [comment]
I confirm the delete issue @nLight mentioned

File: plugins/services/src/js/components/forms/MultiContainerVolumesFormSection.js [hunk 1]
```diff
@@ -95,8 +117,21 @@ class MultiContainerVolumesFormSection extends Component {
...
+      if (
+        volumes.type === VolumeConstants.type.unknown ||
+        volumes.type === VolumeConstants.type.dss
+      ) {
+        return (
+          <FormGroupContainer
+            key={key}
+            onRemove={this.props.onRemoveItem.bind(this, {
+              value: key,
+              path: "volumes"
```
Comment (Poltergeist) [inline plugins/services/src/js/components/forms/MultiContainerVolumesFormSection.js]: This means the remove button will not work. @nLight found this. I will fix it.

File: plugins/dss/hooks.js [hunk 2]
```diff
@@ -127,6 +303,14 @@ module.exports = {
       "CreateService:SingleContainerVolumes:UnknownVolumes"
     );
 
+    MountService.MountService.registerComponent(
+      getDSSInput,
```
Comment (nLight) [inline plugins/dss/hooks.js]: Those are components, name suggestion `DSSInput`

File: plugins/dss/hooks.js [hunk 3]
```diff
@@ -127,6 +303,14 @@ module.exports = {
...
+      "CreateService:MultiContainerVolumes:UnknownVolumes"
+    );
+    MountService.MountService.registerComponent(
+      getVolumeMountTypes,
```
Comment (nLight) [inline plugins/dss/hooks.js]: Those are components, name suggestion `VolumeMountTypesSelect`

File: plugins/services/src/js/components/forms/MultiContainerVolumesFormSection.js [hunk 4]
```diff
@@ -74,6 +76,26 @@ class MultiContainerVolumesFormSection extends Component {
     });
   }
 
+  getUnknownVolumeConfig(volumes, key, offset) {
```
Comment (nLight) [inline plugins/services/src/js/components/forms/MultiContainerVolumesFormSection.js]: Wanted to trigger that view with the app config but instead I'm getting an empty box

screen shot 2018-01-09 at 16 13 24


```
{
  "id": "/asd",
  "containers": [
    {
      "name": "container-1",
      "resources": {
        "cpus": 0.1,
        "mem": 128
      },
      "image": {
        "id": "alpine",
        "kind": "DOCKER"
      },
      "volumeMounts": [
      ]
    }
  ],
  "scaling": {
    "instances": 1,
    "kind": "fixed"
  },
  "networks": [
    {
      "mode": "host"
    }
  ],
  "volumes": [
    {
      "name": "asdf",
      "unknownPath": "sdf"
    }
  ],
  "fetch": [],
  "scheduling": {
    "placement": {
      "constraints": []
    }
  }
}
```

File: plugins/services/src/js/components/forms/MultiContainerVolumesFormSection.js [hunk 5]
```diff
@@ -74,6 +76,26 @@ class MultiContainerVolumesFormSection extends Component {
...
+      <MountService.Mount
+        type="CreateService:MultiContainerVolumes:UnknownVolumes"
+        volumes={volumes}
+        index={key}
+        offset={offset}
+        data={this.props.data}
+        errors={this.props.errors}
+      >
+        <FieldLabel>
+          Unable to edit this Volume{" "}
```
Comment (nLight) [inline plugins/services/src/js/components/forms/MultiContainerVolumesFormSection.js]: Why do we need the empty space at the end?

Comment (nLight) [reply plugins/services/src/js/components/forms/MultiContainerVolumesFormSection.js]: I still believe we don't need this extra trailing space

Comment (Poltergeist) [reply plugins/services/src/js/components/forms/MultiContainerVolumesFormSection.js]: You are right. At least I do not see any more an issue there...
removed the space

File: plugins/services/src/js/components/forms/MultiContainerVolumesFormSection.js (hunk 4 above)
Comment (Poltergeist) [reply plugins/services/src/js/components/forms/MultiContainerVolumesFormSection.js]: This one is fixed.

File: plugins/dss/hooks.js [hunk 6]
```diff
@@ -15,10 +15,196 @@ import FormGroupHeading from "#SRC/js/components/form/FormGroupHeading";
...
+  );
+}
+
+function DSSInput(props) {
+  const { volumes, index, offset } = props;
+  if (volumes.type !== VolumeConstants.type.dss) {
+    return (
+      <div>
+        <FieldLabel>
+          Unable to edit this Volume
```
Comment (ahoskins) [inline plugins/dss/hooks.js]: Just btw, VolumesFormSection puts a space after this, should probably remove there too for consistent: https://github.com/dcos/dcos-ui/blob/master/plugins/services/src/js/components/forms/VolumesFormSection.js#L296

File: plugins/dss/hooks.js [hunk 7]
```diff
@@ -15,10 +15,196 @@ import FormGroupHeading from "#SRC/js/components/form/FormGroupHeading";
...
+            type="CreateService:MultiContainerVolumes:Types"
+            volumes={volumes}
+            index={index}
+          >
+            <FieldSelect
+              name={`volumeMounts.${index}.type`}
+              value={volumes.type}
+            >
+              <option value="">Select...</option>
+              <option value={VolumeConstants.type.host}>Host Volume</option>
```
Comment (ahoskins) [inline plugins/dss/hooks.js]: Would be nice to use these constants in the VolumesFormSection: https://github.com/dcos/dcos-ui/blob/master/plugins/services/src/js/components/forms/VolumesFormSection.js#L350

File: plugins/dss/hooks.js (hunk 6 above)
Comment (Poltergeist) [reply plugins/dss/hooks.js]: Yes you are right I will do that in another PR.

**Comment by Poltergeist** [comment]
Alfred please test this.

**Comment by nLight** [comment]
System tests are green here 💥

**Comment by Poltergeist** [comment]
All is green.! 🏆

**Pull request merged**